NickvisionTubeConverter.GNOME/Views/AddDownloadDialog.cs

```diff
@@ -41,39 +41,183 @@ public partial class AddDownloadDialog : Adw.MessageDialog
     private delegate void GAsyncReadyCallback(nint source, nint res, nint user_data);
 
     [LibraryImport("libadwaita-1.so.0", StringMarshalling = StringMarshalling.Utf8)]
-    private static partial void gtk_file_dialog_save(nint dialog, nint parent, nint cancellable, GAsyncReadyCallback callback, nint user_data);
+    private static partial void gtk_file_dialog_select_folder(nint dialog, nint parent, nint cancellable, GAsyncReadyCallback callback, nint user_data);
 
     [LibraryImport("libadwaita-1.so.0", StringMarshalling = StringMarshalling.Utf8)]
-    private static partial nint gtk_file_dialog_save_finish(nint dialog, nint result, nint error);
+    private static partial nint gtk_file_dialog_select_folder_finish(nint dialog, nint result, nint error);
```

Comment:
We still need `save` for selecting a path for single media as was discussed in matrix.

Reply:
The single media works the same way as a playlist just with one download instead of 5, similar to WinUI if you look at that. So there is no need for `save`.

Reply:
Well, I thought will continue to use "Save path" for single media and use "Save folder" only for playlists. But if you want to revert it, okay. No checks of file existence again tho.

Reply:
The problem is it would be too complicated in the backend to separate `save path` vs `save folder` when dealing with single video vs playlist. We could add another row to the AddDownloadDialog that's a simple toggle called `Overwrite Existing Files` which we can pass to yt-dlp per download whether to overwrite files or not.

Because remember, if i'm downloading an `mp3` there will first be an `mp4` downloaded and then a `mp3`. So with the `save` dialog if I was to agree to overwriting the `mp3` file, the `mp4` file will be implicitly overwritten tho which the user might not want.

So i think this toggle might be the best solution...

Reply:
From yt-dlp:
```
paths:             Dictionary of output paths. The allowed keys are 'home'
                       'temp' and the keys of OUTTMPL_TYPES (in utils.py)
```
So we can have all temporary files in separate folder and not worry about overriding in cases such as you provided. Then only the target file overriding should be considered, and so we don't need a toggle, only checking with `File.Exists(folder+name+ext)` and showing a warning if the folder is opened through portal.

Reply:
so if said file exists, we will have a label saying something like `There are files in the save folder that conflict with the titles of some videos to be downloaded. These files will be overwritten. To prevent this, change the title of the video to download.`

Reply:
> so if said file exists, we will have a label saying something like `There are files in the save folder that conflict with the titles of some videos to be downloaded. These files will be overwritten. To prevent this, change the title of the video to download.`

No. We should set `temp` in `paths` dict to something like `/tmp/Nickvision/TubeConverter` (using available dotnet variable to get the equivalent of `/tmp` on Windows) so all files except target file will be placed there.
Then for target file only we should produce 2 checks:
1. `File.Exists()`
2. If a folder is not available for flatpak and so accessible though portal. We already have this check to show a warning that the folder will not be remembered, and this warning should be extended to provide information that it's impossible to check if the file exists (because `File.Exists()` will always return `false` in portalled folder, since the actual content of the real folder can't be read).

Reply:
> We should set temp in paths dict to something like /tmp/Nickvision/TubeConverter (using available dotnet variable to get the equivalent of /tmp on Windows

We have the config path where we could create a temp folder there and use, to be consistent across platforms.

Reply:
I committed an implementation of the overwrite toggle in the add download dialog. I feel like it would be a lot easier from a user's perspective to have a toggle switched on and off depending on whether or not they will want to overwrite files and pass that logic to yt-dlp, instead of going through flatpak permissions and warning the user with all that complicated sandbox stuff. According to yt-dlp, if overwrite is false it won't override both the mp4 and mp3 files used for downloading a mp3 file, as described in the previous scenario.

Check it out and let me know what you think, hope we can find common ground on that :)